--- a/src/pamhelper/Makefile.am
+++ b/src/pamhelper/Makefile.am
@@ -5,9 +5,10 @@
 
 SUBDIRS =
 
-pamhelperdir = $(datadir)/cinnamon-screensaver/pamhelper
+pamhelperdatadir = $(datadir)/cinnamon-screensaver/pamhelper
+pamhelperdir = $(libexecdir)/pamhelper
 
-pamhelper_PYTHON = \
+pamhelperdata_PYTHON = \
     authClient.py \
     $(NULL)
 
--- a/src/pamhelper/authClient.py
+++ b/src/pamhelper/authClient.py
@@ -2,6 +2,7 @@
 
 from gi.repository import Gio, GObject, GLib, CScreensaver
 
+import config
 import re
 import os
 import signal
@@ -37,7 +38,7 @@
         try:
             helper_path = None
             architecture = platform.machine()
-            paths = ["/usr/lib", "/usr/lib/cinnamon-screensaver", "/usr/libexec"]
+            paths = [config.libexecdir, "/usr/lib", "/usr/lib/cinnamon-screensaver", "/usr/libexec"]
 
             # On x86 archs, iterate through multiple paths
             # For instance, on a Mint i686 box, the path is actually /usr/lib/i386-linux-gnu
